console-subscriber/src/aggregator/mod.rs

```diff
@@ -769,50 +783,105 @@ impl Aggregator {
                 self.new_poll_ops.push(poll_op);
             }
 
-            Event::StateUpdate {
-                resource_id,
-                update,
-                ..
-            } => {
-                let resource_id = self.ids.id_for(resource_id);
-                if let Some(mut stats) = self.resource_stats.update(&resource_id) {
-                    let field_name = match update.field.name.clone() {
-                        Some(name) => name,
+            Event::StateUpdate { target, update, .. } => {
+                let target_id = self.ids.id_for(target.target_id);
+                let mut targets_to_update = vec![(target_id, target.target_type.clone())];
+
+                fn update_entry(
+                    mut e: std::collections::hash_map::Entry<'_, FieldKey, Attribute>,
+                    upd: &AttributeUpdate,
+                ) {
+                    match e {
+                        Entry::Occupied(ref mut attr) => {
+                            update_attribute(attr.get_mut(), upd);
+                        }
+                        Entry::Vacant(attr) => {
+                            attr.insert(upd.clone().into());
+                        }
+                    }
+                }
+
+                match target.target_type {
+                    TargetType::Resource => {
+                        if let Some(parent) = self
+                            .resources
+                            .get(&target_id)
+                            .and_then(|r| r.parent_id)
+                            .and_then(|parent_id| self.resources.get(&parent_id))
+                            .filter(|parent| parent.inherit_child_attrs)
+                        {
+                            targets_to_update.push((parent.id, TargetType::Resource));
+                        }
+                    }
+                    TargetType::AsyncOp => {
+                        if let Some(parent) = self
+                            .async_ops
+                            .get(&target_id)
+                            .and_then(|ao| ao.parent_id)
+                            .and_then(|parent_id| self.async_ops.get(&parent_id))
```

Comment:
similarly, i think this could be
```suggestion
nit, take it or leave it: i think we could combine this into one `and_then` like:
```suggestion
                            .and_then(|r| self.async_ops.get(r.parent_id.as_ref()?))
```
```